src/DynamicRupture/Output/Builders/ElementWiseBuilder.h

```diff
@@ -34,54 +38,135 @@ class ElementWiseBuilder : public ReceiverBasedOutputBuilder {
+                         << "Number of sub-triangles:" << numSubTriangles;
 
-    // get arrays of elements and vertices from the meshReader
-    const auto& faultInfo = meshReader->getFault();
-    const auto& elementsInfo = meshReader->getElements();
-    const auto& verticesInfo = meshReader->getVertices();
+      // get arrays of elements and vertices from the meshReader
+      const auto& faultInfo = meshReader->getFault();
+      const auto& elementsInfo = meshReader->getElements();
+      const auto& verticesInfo = meshReader->getVertices();
 
-    // iterate through each fault side
-    for (size_t faceIdx = 0; faceIdx < numFaultElements; ++faceIdx) {
+      // iterate through each fault side
+      for (size_t faceIdx = 0; faceIdx < numFaultElements; ++faceIdx) {
 
-      // get a global element ID for the current fault face
-      const auto& fault = faultInfo[faceIdx];
-      const auto elementIdx = fault.element;
+        // get a global element ID for the current fault face
+        const auto& fault = faultInfo[faceIdx];
+        const auto elementIdx = fault.element;
 
-      if (elementIdx >= 0) {
-        const auto& element = elementsInfo[elementIdx];
+        if (elementIdx >= 0) {
+          const auto& element = elementsInfo[elementIdx];
 
-        // store coords of vertices of the current ELEMENT
-        constexpr size_t numVertices{4};
-        std::array<const double*, numVertices> elementVerticesCoords{};
-        for (size_t vertexIdx = 0; vertexIdx < numVertices; ++vertexIdx) {
-          auto globalVertexIdx = element.vertices[vertexIdx];
-          elementVerticesCoords[vertexIdx] = verticesInfo[globalVertexIdx].coords;
+          // store coords of vertices of the current ELEMENT
+          constexpr size_t numVertices{4};
+          std::array<const double*, numVertices> elementVerticesCoords{};
+          for (size_t vertexIdx = 0; vertexIdx < numVertices; ++vertexIdx) {
+            auto globalVertexIdx = element.vertices[vertexIdx];
+            elementVerticesCoords[vertexIdx] = verticesInfo[globalVertexIdx].coords;
+          }
+
+          const auto faceSideIdx = fault.side;
+
+          // init reference coordinates of the fault face
+          ExtTriangle referenceTriangle = getReferenceTriangle(faceSideIdx);
+
+          // init global coordinates of the fault face
+          ExtTriangle globalFace = getGlobalTriangle(faceSideIdx, element, verticesInfo);
+
+          faultRefiner->refineAndAccumulate(
+              {elementwiseParams.refinement, static_cast<int>(faceIdx), faceSideIdx, elementIdx},
+              std::make_pair(globalFace, referenceTriangle));
         }
+      }
+
+      // retrieve all receivers from a fault face refiner
+      outputData->receiverPoints = faultRefiner->moveAllReceiverPoints();
+      faultRefiner.reset(nullptr);
+    } else {
+      const auto order = elementwiseParams.vtkorder;
+      logInfo(localRank) << "Initializing fault field output. Order:" << order;
```

Comment:
The same with this log output, either we remove this for both and let the general manner VTK output take care of the logs but still print the number of sub triangles used for fault output for both the cases

Reply:
Ok; thanks. Yeah, I'll remove that one.

Regarding the triangle subsampling: the next step would be to extend that as well to high-order (but not quite now, yet). The message will stay for now; transitioning the old IO system to the new one is still a TODO.